main/lsp/BUILD

```diff
@@ -1,3 +1,11 @@
+config_setting(
+    name = "enable_rubyfmt",
+    values = {
+        "define": "rubyfmt=true",
+    },
+    visibility = ["//visibility:public"],
+)
```

Comment:
Since you're deleting the build-time dependency on e.g. a rust toolchain and librubyfmt, there's no real reason why we need an `enable_rubyfmt` config setting anymore. We could build support for document formatting unconditionally, and gate it behind a normal (runtime) command line flag, rather than a build-time flag. Note: that it appears such a flag already exists and is in use.

Let's go ahead and just delete this config setting, and also delete the `noop.cc` formatting implementation, as neither are required anymore.